go/vt/sqlparser/sql.y

```diff
@@ -769,6 +770,10 @@ compare:
   {
     $$ = NullSafeEqualStr
   }
+| COLLATE
```

Comment:
This shouldn't be required. Can you remove it and see if things work as intended?